app/controllers/concerns/reports/csv_export_controller.rb

```diff
@@ -0,0 +1,27 @@
+module Reports
+
+  module CsvExportController
+
+    def export_all
+      generate_report_data_csv
+    end
+
+    private
+
+    def email_to_address
+      params[:email_to_address].presence || current_user.email
+    end
+
+    def generate_report_data_csv
+      ExportRawReportMailer.delay.raw_report_email(email_to_address, raw_report)
```

Comment:
This is kind of a weird one, I think. It probably tries to convert it to Global ID, but we actually need to drop in the full report. I think we're generating the report in the app, and then just delaying the email sending. Ideally, we would delay the report creation. If it's too complicated, it might not be worth it.